File: gencode/java/udmi/schema/DiscoveredNetwork.java

package udmi.schema;

import java.util.HashMap;
import javax.annotation.processing.Generated;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;


/**
 * Discovered Network
 * <p>
 * 
 * 
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
    "families"
})
@Generated("jsonschema2pojo")
public class DiscoveredNetwork {

    @JsonProperty("families")
    public HashMap<String, DiscoveredNetworkFamily> families;

    @Override
    public int hashCode() {
        int result = 1;
        result = ((result* 31)+((this.families == null)? 0 :this.families.hashCode()));
        return result;
    }

    @Override
    public boolean equals(Object other) {
        if (other == this) {
            return true;
        }
        if ((other instanceof DiscoveredNetwork) == false) {
            return false;
        }
        DiscoveredNetwork rhs = ((DiscoveredNetwork) other);
        return ((this.families == rhs.families)||((this.families!= null)&&this.families.equals(rhs.families)));
    }

}

File: gencode/python/udmi/schema/discovered_network.py
"""Generated class for discovered_network.json"""
from .discovered_network_family import DiscoveredNetworkFamily


class DiscoveredNetwork:
  """Generated schema class"""

  def __init__(self):
    self.families = None

  @staticmethod
  def from_dict(source):
    if not source:
      return None
    result = DiscoveredNetwork()
    result.families = DiscoveredNetworkFamily.map_from(source.get('families'))
    return result

  @staticmethod
  def map_from(source):
    if not source:
      return None
    result = {}
    for key in source:
      result[key] = DiscoveredNetwork.from_dict(source[key])
    return result

  @staticmethod
  def expand_dict(input):
    result = {}
    for property in input:
      result[property] = input[property].to_dict() if input[property] else {}
    return result

  def to_dict(self):
    result = {}
    if self.families:
      result['families'] = DiscoveredNetworkFamily.expand_dict(self.families) # 2
    return result
